### ISSUE TITLE
Compiler warnings in Classic Driver

### ISSUE THREAD

**Issue by jhamman** [UW-Hydro/VIC#430]
These just popped up recently on the develop branch:

```bash
$ gcc --version
Configured with: --prefix=/Applications/Xcode.app/Contents/Developer//usr --with-gxx-include-dir=/usr/include/c++/4.2.1
Apple LLVM version 7.3.0 (clang-703.0.29)
Target: x86_64-apple-darwin15.4.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
```

```make
./src/read_soilparam.c:84:9: warning: variable 'temp' is used uninitialized whenever 'if' condition is false [-Wsometimes-uninitialized]
    if (!(*MODEL_DONE) && (*RUN_MODEL)) {
        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./src/read_soilparam.c:934:12: note: uninitialized use occurs here
    return temp;
           ^~~~
./src/read_soilparam.c:84:5: note: remove the 'if' if its condition is always true
    if (!(*MODEL_DONE) && (*RUN_MODEL)) {
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./src/read_soilparam.c:84:9: warning: variable 'temp' is used uninitialized whenever '&&' condition is false [-Wsometimes-uninitialized]
    if (!(*MODEL_DONE) && (*RUN_MODEL)) {
        ^~~~~~~~~~~~~~
./src/read_soilparam.c:934:12: note: uninitialized use occurs here
    return temp;
           ^~~~
./src/read_soilparam.c:84:9: note: remove the '&&' if its condition is always true
    if (!(*MODEL_DONE) && (*RUN_MODEL)) {
        ^~~~~~~~~~~~~~~~~
./src/read_soilparam.c:64:5: note: variable 'temp' is declared here
    soil_con_struct          temp;
    ^
2 warnings generated.
```

**Comment by bartnijssen** [comment]
@jhamman: Are these still there? If not, just close out.

**Comment by jhamman** [comment]
yes, I fixed these in #552.
